pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -457,10 +465,57 @@ func (o *JobRunTestCaseAnalyzerOptions) findJobRunsWithRetry(ctx context.Context
 	}
 }
 
+func (o *JobRunTestCaseAnalyzerOptions) loadStaticJobRuns(ctx context.Context, jobName string, jobRunLocator jobrunaggregatorlib.JobRunLocator) ([]jobrunaggregatorapi.JobRunInfo, error) {
+	var outputRuns []jobrunaggregatorapi.JobRunInfo
+	for _, jobRun := range o.staticJobRunIdentifiers {
+		if jobRun.JobName != jobName {
+			continue
+		}
+
+		jobRun, err := jobRunLocator.FindJob(ctx, jobRun.JobRunID)
+		if err != nil {
+			return nil, err
+		}
+		if jobRun != nil {
+			outputRuns = append(outputRuns, jobRun)
+		}
+	}
+	return outputRuns, nil
+}
+
+func (o *JobRunTestCaseAnalyzerOptions) loadStaticJobs() []jobrunaggregatorapi.JobRow {
+	rows := make([]jobrunaggregatorapi.JobRow, 0)
+	uniqueNames := sets.Set[string]{}
+
+	for _, r := range o.staticJobRunIdentifiers {
+		// only one row per unique job name
+		if !uniqueNames.Has(r.JobName) {
+			// we only care about returning JobName
+			rows = append(rows, jobrunaggregatorapi.JobRow{JobName: r.JobName})
+			uniqueNames.Insert(r.JobName)
+		}
+	}
+
+	return rows
+}
+
+func (o *JobRunTestCaseAnalyzerOptions) SetRelatedJobRuns(jobRunIDs []jobrunaggregatorlib.JobRunIdentifier) {
+	o.staticJobRunIdentifiers = jobRunIDs
+}
+
 // GetRelatedJobRuns gets all related job runs for analysis
 func (o *JobRunTestCaseAnalyzerOptions) GetRelatedJobRuns(ctx context.Context) ([]jobrunaggregatorapi.JobRunInfo, error) {
 	var jobRunsToReturn []jobrunaggregatorapi.JobRunInfo
-	jobs, err := o.jobGetter.GetJobs(ctx)
+	var jobs []jobrunaggregatorapi.JobRow
+	var err error
+
+	// allow for the list of ids to be passed in
+	if len(o.staticJobRunIdentifiers) > 0 {
```

Comment:
You already have the job names and job ids. It is probably better to create a common library for both analyzers to GetRelatedJobRunsFromStaticIdentifiers.

Reply:
GetRelatedJobRuns is returning []jobrunaggregatorapi.JobRunInfo which we need jobLocators to fetch the JobRunInfo and those are constructed differently for jobrunaggregatoranalyzer and jobruntestcaseanalyzer.

These changes also allow us to pass in the list of jobs via command line, potentially for aggregating results from tools like gangway, so where we are reusing that code now we would have to duplicate some of it for the util.GetRelatedJobRunsFromStaticIdentifiers and pass in joblocator(s).

I see what you are saying but I think the implementation wouldn't be straightforward and might lead to even more duplicated code.

Reply:
I was thinking of skipping the locator and matchers. You already have the names and the corresponding IDs, we can do a straight read (maybe using ReadJobRunFromGCS, common for both analyzers) instead of gcs search (from startingJobRunID to endingJobRunID). That should further reduce our gcs usage. 

But we can also leave that for future enhancement.

Reply:
I believe that is what it is, effectively, doing.  Just via the locator.

https://github.com/neisw/ci-tools/blob/e1d979b851238f356771617eaf1cd45e376df057/pkg/jobrunaggregator/jobrunaggregatoranalyzer/analyzer.go#L81

and 

https://github.com/neisw/ci-tools/blob/e1d979b851238f356771617eaf1cd45e376df057/pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go#L475

Call 

https://github.com/neisw/ci-tools/blob/e1d979b851238f356771617eaf1cd45e376df057/pkg/jobrunaggregator/jobrunaggregatorlib/jobrun_locator.go#L88

Reply:
I could rename it from FindJob to ReadJob or something if that makes it clearer.  I think I went with FindJob since the existing api was FindRelatedJobs but I'm not partial to the api name.

Reply:
Yes, as of now, this is used on the per job level. You still had to fit your code into the existing framework, like for test case analyzer, you had to get the list of job names before going to the per job level. But you already have the list of job names and the their corresponding job run IDs in both analyzers. If you have a common function to get related job runs from jobNames and their corresponding Run IDs, you could call that function and return from here, without using the logic that is unique to that particular analyzer. Let's sync on a huddle if that is easier.

Reply:
Reviewed with Ken, the request was to see if we could create a generic locator that could handle multiple jobs.  The locator requires a prefix that is often generic in the form of  "logs/" + JobName.

However in the case of PR jobs it is nondeterministic and needs to be passed in 
`fs.Var(&jobGCSPrefixSlice{&f.JobGCSPrefixes}, "explicit-gcs-prefixes", "a list of gcs prefixes for jobs created for   payload. Only used by per PR payload promotion jobs. The format is comma-separated elements, each consisting of job name and gcs prefix separated by =, like openshift-machine-config-operator=3028-ci-4.11-e2e-aws-ovn-upgrade~logs/openshift-machine-config-operator-3028-ci-4.11-e2e-aws-ovn-upgrade")`

After review we decided it is not worth the effort to attempt the refactor currently.